### PR TITLE
quarkus jpa repository pattern test

### PR DESCRIPTION
**Pull request by arolfes**

### CODE REVIEW

File: examples/data-jpa-repository/src/main/kotlin/cnt/jpa/domain/Movie.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+data class Movie(
+
+    @Id
+    val id: Long,
+
+    val title: String,
+
+    val publishYear: Int
+
+)
```
Comment (slu-it) [inline examples/data-jpa-repository/src/main/kotlin/cnt/jpa/domain/Movie.kt:16]: Why so many blank lines? Wouldn't
```
@Entity
data class Movie(
    @Id
    val id: Long,
    val title: String,
    val publishYear: Int
)
```
be more readable?

File: examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        val movie4 = Movie(1, "Star Wars Episode IV - A New Hope", 1977)
+        movieRepository.persist(movie4)
+        val movieRougeOne = Movie(2, "Rogue One: A Star Wars Story", 2016)
+        movieRepository.persist(movieRougeOne)
+        val movie9 = Movie(3, "Star Wars Episode IX: The Rise of Skywalker", 2019)
+        movieRepository.persist(movie9)
+
+
+        val foundEntities = movieRepository.findByTitle("Rogue One: A Star Wars Story")
+        Assertions.assertThat(foundEntities)
```
Comment (slu-it) [inline examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt:43]: static import?

File: examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    fun `should persist and find a movie when persist and find method called`() {
+        val id = 1L
+        val movie = Movie(id, "Star Wars Episode IV - A New Hope", 1977)
+
+        movieRepository.persist(movie)
+
+        val foundEntity = movieRepository.findById(id)
+        Assertions.assertThat(foundEntity).isEqualTo(movie)
```
Comment (slu-it) [inline examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt:28]: static import?

File: examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+@TestProfile(H2TestProfile::class)
+@QuarkusTestResource(H2DatabaseTestResource::class)
+class MovieRepositoryWithH2Test {
+
+    @Inject
+    lateinit var movieRepository: MovieRepository
+
+    @BeforeEach
+    fun clearDatabase() {
+        movieRepository.deleteAll()
```
Comment (slu-it) [inline examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt:23]: Question: if the test itself is transactional (rollback at the end of each test like in Spring?) is this cleanup strictly necessary?

File: examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+@TransactionalQuarkusTest
+class MovieRepositoryWithPostgresTest {
+
+    @Inject
+    lateinit var movieRepository: MovieRepository
+
+    @BeforeEach
+    fun clearDatabase() {
+        movieRepository.deleteAll()
```
Comment (slu-it) [inline examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithPostgresTest.kt:17]: see question from MovieRepositoryWithH2Test.kt

File: examples/data-jpa-repository/src/main/kotlin/cnt/jpa/domain/Movie.kt (hunk 1 above)
Comment (arolfes) [reply examples/data-jpa-repository/src/main/kotlin/cnt/jpa/domain/Movie.kt:16]: i love it ;-)

File: examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt (hunk 4 above)
Comment (arolfes) [reply examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt:23]: the rollback does not happen automatically

Comment (arolfes) [reply examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt:23]: I understand it like this:
with the annotation `@TransactionalQuarkusTest` each method is declared as transactional which means that after each test method a full commit and flush to database happens.
without the transactional annotation, i would only test the entity manager behavior, but would not see if the insert really works (for complex inserts)

Comment (slu-it) [reply examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt:23]: Good to know

**Review by slu-it — APPROVED** [review]

**Pull request merged**